datasources/beeper/beeper.go

```diff
@@ -0,0 +1,982 @@
+
+				count++
+			}
+		}
+		rows.Close()
+
+		totalProcessed += count
+		params.Log.Info("Processed users batch",
+			zap.Int("batch_count", count),
+			zap.Int("total_processed", totalProcessed),
+			zap.Int("total_users", totalUsers),
+			zap.Float64("percent_complete", float64(totalProcessed)/float64(totalUsers)*100))
+
+		// If we got fewer rows than the batch size, we're done
+		if count < batchSize {
+			break
+		}
+
+		// Move to the next batch
+		offset += batchSize
+	}
+
+	params.Log.Info("Completed users import", zap.Int("total_users", totalProcessed))
+	return nil
+}
+
+// Helper functions
+func checkTables(src string) (bool, error) {
+	tempDir, err := os.MkdirTemp("", "beeper_import_*")
+	if err != nil {
+		return false, fmt.Errorf("creating temp directory: %w", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	db, err := openDB(src, tempDir)
+	if err != nil {
+		return false, fmt.Errorf("opening database: %w", err)
+	}
+	defer db.Close()
+
+	// Check for message-related tables
+	tables, err := getTableNames(db)
+	if err != nil {
+		return false, err
+	}
+
+	// Look for tables that suggest this is a Beeper database
+	return containsTable(tables, "mx_room_messages") ||
+		containsTable(tables, "threads") ||
+		containsTable(tables, "users"), nil
+}
+
+// getTableNames returns all table names in the database
+func getTableNames(db *sql.DB) ([]string, error) {
+	rows, err := db.Query("SELECT name FROM sqlite_master WHERE type='table'")
+	if err != nil {
+		return nil, fmt.Errorf("querying table names: %w", err)
+	}
+	defer rows.Close()
+
+	var tables []string
+	for rows.Next() {
+		var name string
+		if err := rows.Scan(&name); err != nil {
+			return nil, fmt.Errorf("scanning table name: %w", err)
+		}
+		tables = append(tables, name)
+	}
+
+	return tables, nil
+}
+
+// getTableColumns returns all column names for a table
+func getTableColumns(db *sql.DB, tableName string) ([]string, error) {
+	query := fmt.Sprintf("PRAGMA table_info(%s)", tableName)
+	rows, err := db.Query(query)
+	if err != nil {
+		return nil, fmt.Errorf("getting table columns: %w", err)
+	}
+	defer rows.Close()
+
+	var columns []string
+	for rows.Next() {
+		var id int
+		var name, dataType string
+		var notNull, pk int
+		var dfltValue interface{}
+		if err := rows.Scan(&id, &name, &dataType, &notNull, &dfltValue, &pk); err != nil {
+			return nil, fmt.Errorf("scanning column info: %w", err)
+		}
+		columns = append(columns, name)
+	}
+
+	return columns, nil
+}
+
+// findColumnIndex returns the index of a column in the columns slice, or -1 if not found
+func findColumnIndex(columns []string, name string) int {
+	for i, col := range columns {
+		if col == name {
+			return i
+		}
+	}
+	return -1
+}
+
+func containsTable(tables []string, table string) bool {
+	for _, t := range tables {
+		if t == table {
+			return true
+		}
+	}
+	return false
+}
+
+// toString converts an interface to a string
+func toString(value interface{}) string {
```

Comment:
```suggestion
func toString(value any) string {
```

(Replace all instances of `interface{}` with `any`)